crates/block-producer/src/batch_builder/mod.rs

```diff
@@ -1,202 +1,195 @@
-use std::{cmp::min, collections::BTreeSet, sync::Arc, time::Duration};
+use std::{num::NonZeroUsize, ops::Range, time::Duration};
 
-use async_trait::async_trait;
-use miden_objects::{notes::NoteId, transaction::OutputNote};
-use tokio::time;
+use rand::Rng;
+use tokio::{task::JoinSet, time};
 use tracing::{debug, info, instrument, Span};
 
-use crate::{block_builder::BlockBuilder, ProvenTransaction, SharedRwVec, COMPONENT};
+use crate::{
+    domain::transaction::AuthenticatedTransaction,
+    mempool::{BatchJobId, SharedMempool},
+    COMPONENT, SERVER_BUILD_BATCH_FREQUENCY,
+};
 
-#[cfg(test)]
-mod tests;
+// FIXME: fix the batch builder tests.
+// #[cfg(test)]
+// mod tests;
```

Comment:
Similar question as above: should we delete this or re-enable it?

Reply:
I've opted to remove this.